lib/proxy/helpers.rb

```diff
@@ -25,7 +25,8 @@ def log_halt code=nil, exception=nil
 
   # read the HTTPS client certificate from the environment and extract its CN
   def https_cert_cn
-    certificate_raw = request.env['SSL_CLIENT_CERT'].to_s
+    ssl_client_cert_header = Proxy::SETTINGS.ssl_client_cert_header || 'SSL_CLIENT_CERT'
+    certificate_raw = request.env[ssl_client_cert_header].to_s.gsub(/   /, "\n")
```

Comment:
From my point of view, creating another function, just to make sure the certificate is in valid format is unnecessary. We need to skip one space case, because that one space is included in PEM header. Maybe we can replace ` ` with `\s`.